internal/controllers/machineset/machineset_delete_policy.go

```diff
@@ -150,5 +150,9 @@ func isMachineHealthy(machine *clusterv1.Machine) bool {
 	if nodeHealthyCondition != nil && nodeHealthyCondition.Status != corev1.ConditionTrue {
 		return false
 	}
+	healthCheckCondition := conditions.Get(machine, clusterv1.MachineHealthCheckSucceededCondition)
```

Comment:
do we want to align criteria with above?
```nodeHealthyCondition.Status != corev1.ConditionTrue {`

Reply:
I don't think we have to.
For NodeHealthyCondition, a machine is considered unhealthy if the conditions is missing or if it is false (not true)
Instead, for HealthCheckSucceededCondition we should consider unhealthy only machines where the condition exists (is set by MHC)  and it is false.

In other words, it is ok a machine doesn't have the HealthCheckSucceededCondition (e.g. when there is no MHC targeting a machine)

Reply:
I think the "existence" check is the `!= nil` part not were we check for the Status

Reply:
So I think the difference between both checks here is how we treat Unknown?

Reply:
In practice won't make much difference they things are i.e. the mhc doesn't set Unknown explicitly today.
Semantically and to avoid confusion I think we should align both checks. If I were new the project and I see we do `healthCheckCondition.Status == corev1.ConditionFalse {` in one place and `nodeHealthyCondition.Status != corev1.ConditionTrue {` in the other I would definitely try to find out why with no luck because the way we put it is kind of arbitrary.

Reply:
AFAIK MHC never sets this condition to Unknown, but for HealthCheckSucceededCondition I think it is ok to consider a machine HealthCheckSucceededCondition=UnKnown as healthy for the sake of delete prioritization.

With regards to NodeHealthyCondition, I don't have the full context about current state, but  NodeHealthyCondition is set when it fails to get the Node for this Machine by ProviderID, which IMO it is correctly treated as unhealthy.

Reply:
Talked with Fabrizio about it, we'll treat unknown the same way. PR coming up

Reply:
PR: https://github.com/kubernetes-sigs/cluster-api/pull/10763